internal/ioutil/ioutil.go

```diff
@@ -371,15 +357,36 @@ func CopyAligned(w io.Writer, r io.Reader, alignedBuf []byte, totalSize int64, f
 		}
 
 		buf = buf[:nr]
-		var nw int64
-		if len(buf)%DirectioAlignSize == 0 {
-			var n int
+		var (
+			n  int
+			un int
+			nw int64
+		)
+
+		remain := len(buf) % DirectioAlignSize
+		if remain == 0 {
```

Comment:
No need for two writes if platform does not have directio.
```suggestion
		if !disk.ODirectPlatform || remain == 0 {
```

Reply:
this won't even come here.